src/main/java/io/jenkins/plugins/pipeline_log_fluentd_cloudwatch/CloudWatchRetriever.java

```diff
@@ -114,8 +118,9 @@ private OverallLog(ByteBuffer buf, boolean completed, FlowExecutionOwner.Executa
 
         @Override
         public long writeHtmlTo(long start, final Writer w) throws IOException {
-            if (start == 0 && /* would mess up PipelineBridgeTest */ !Main.isUnitTest) {
-                String url = "https://console.aws.amazon.com/cloudwatch/home#logEventViewer:group=" + logGroupName + ";stream=" + logStreamName + ";filter=%257B%2524.build%2520%253D%2520%2522" + buildId + "%2522%257D";
+            if (start == 0 && !Main.isUnitTest) { // would mess up PipelineBridgeTest
+                // TODO will not display agent-generated messages; could use DescribeLogStreams to generate all links but there might be a lot
+                String url = "https://console.aws.amazon.com/cloudwatch/home#logEventViewer:group=" + logGroupName + ";stream=" + logStreamNameBase + "@master;filter=%257B%2524.build%2520%253D%2520%2522" + buildId + "%2522%257D";
```

Comment:
Could we make make this a little easier on the eyes? Or at minimum leave a comment here telling us what `%257B%2524.build%2520%253D%2520%2522` actually is?

```java
private static final String CLOUD_WATCH_LOG_QUERY = "https://console.aws.amazon.com/cloudwatch/home#logEventViewer:group=%s;stream=%s@master;filter=%s";
private static final String CLOUD_WATCH_FILTER_QUERY = "{ $.build = \"%s\" }";

public long writeHtmlTo(long start, final Writer w) throws IOException {
    String urlEncodedFilter = URLEncoder.encode(String.format(CLOUD_WATCH_FILTER_QUERY, buildId), "UTF-8").replaceAll("\\+", "%20");
    urlEncodedFilter = URLEncoder.encode(urlEncodedFilter, "UTF-8").replaceAll("\\+", "%20");

    String nicerURl = String.format(CLOUD_WATCH_LOG_QUERY, logGroupName, logStreamNameBase, urlEncodedFilter);
    ....
}
```